### PR TITLE
`fn rav1d_open::error`: Remove

### PR DESCRIPTION
**Pull request by kkysen**
The null check is removed since the allocation is now infallible.

The "default allocator now having a cookie" check is moved up to use `validate_input!` instead.
I also changed it to return `EINVAL` instead of `ENOMEM` (different from C), since it's an argument being checked, and there's no out-of-memory error.

Since both are before the `Rav1dContext` exists or is (non-default) initialized,
we don't need to call `close_internal` here, which frees up some other API designs.

### CODE REVIEW

File: src/cdef.rs [hunk 1]
```diff
@@ -197,7 +197,7 @@ unsafe fn padding<BD: BitDepth>(
     for y in 0..h {
         let tmp = &mut tmp[(y + 2) * TMP_STRIDE..];
         let src = src + (y as isize * stride);
-        let src = &*src.data.slice::<BD, _>((src.offset.., ..x_end));
+        let src = &*src.data.slice::<BD, _>((src.offset.., ..x_end - 2));
```
Comment (randomPoison) [inline src/cdef.rs:200]: Is this change meant to be part of this PR? It seems unrelated to the rest of the changes.

Comment (kkysen) [reply src/cdef.rs:200]: Whoops, no, I forgot to change the base branch.  That's #1178.

**Review by randomPoison — APPROVED** [review]
